### ISSUE TITLE
usage of repository in requirements.yaml is unclear

### ISSUE THREAD

**Issue by jeff-knurek** [kubernetes/helm#2672]
As per the documentation on using requirements.yaml, one can do something along the lines of:
```
# parentchart/requirements.yaml
dependencies:
  - name: subchart
    repository: http://localhost:10191
    version: 0.1.0
```
----
When adding a subchart from the official repository: https://github.com/kubernetes/charts
> This is the default repository for Helm and is installed by default. 

There should be no need for `repository: http://localhost:10191` in this case, since the default repository should be searched. However, not including it leads to an (unhelpful) error: 
```
Error: no repository definition for . Try 'helm repo add'
```
----
Using the repository name however:
```
# parentchart/requirements.yaml
dependencies:
  - name: subchart
    repository: stable
    version: 0.1.0
```
also fails:
```Error: no repository definition for stable. Try 'helm repo add'```

----

The only option then is to try `helm repo list` and include the URL provided for stable. 
```
NAME  	URL
stable	https://kubernetes-charts.storage.googleapis.com
local 	http://127.0.0.1:8879/charts
```

----

In summary, I think this amounts to 2 Feature Requests, and maybe some better documentation/error messaging.
Feature requests:

1. Install from `default`/`stable` repository if a repository is not provided
1. Use `Name` of repo instead of URL (easier to make one change to update all charts)

**Comment by technosophos** [comment]
For (2) above, the problem is that a repo _name_ is a local attribute (local to however you ran `helm repo add`). URL is the canonical reference for a repository.

So, for example, if I do 

```
$ helm repo add myRepo https://matt.example.com/
```

And you do:

```
$helm repo add myRepo https://jeff.example.com/
```

You and I will both have a repo named `myRepo`, but yours is different than mine. The chart must be independent of the local config. Otherwise, when you and I each install the same chart, we'll get different results.

That impacts (1), since we can't assume (a) that everyone has `stable` installed (and pointed to the same URL), or (b) that we all have the same one set as default.

So the root of the problem is really that a chart needs to guarantee independence from the system on which it is running, when it comes to naming.

**Comment by Noah-Huppert** [comment]
Perhaps the error message and docs should be clearer to indicate the value needs to be a url.

**Comment by iamrandys** [comment]
👍

**Comment by davidnewman** [comment]
@Noah-Huppert @jeff-knurek 

You can actually do the following and it works with a repository alias just fine:

`repository: '@somerepoalias'`

This is actually critical because if you have done some work to put your private helm repository behind something like HTTP Basic auth, then using the actual URL doesn't make any sense.

I think the alias feature is OK if you're using a private repository because you can enforce the alias in some way.

**Comment by benkeil** [comment]
This should be added to the documentation.

**Comment by davidnewman** [comment]
@benkeil Well, it's "kind of" in the documentation. It just isn't very well organized.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/requirements.md

This basically lets you know that you can use `@` or `alias://` for the URL field of the requirements.yaml. 

It is not what I would call ideally documented. I may take a stab at updating the docs, but at the same time, it would be hard to do that since a lot of the other documentation is spread out, and not what I would consider "discoverable".
